lib/gollum-lib/markups.rb

```diff
@@ -1,20 +1,58 @@
 # ~*~ encoding: utf-8 ~*~
 module Gollum
+  module MarkupRegisterUtils
+    # Check if a gem exists. This implementation requires Gem::Specificaton to
+    # be filled.
+    def gem_exists? name
+      return Gem::Specification.find {|x| x.name == name} != nil
+    end
+
+    # Check if an executable exists. This implementation comes from
+    # stackoverflow question 2108727.
+    def executable_exists? name
+      exts = ENV["PATHEXT"] ? ENV["PATHEXT"].split(";") : [""]
```

Comment:
It shall be something like ".exe" on windows, and nothing on Linux/macOS. But I have not tried that on windows, just copied that from the SO answer.